### PR TITLE
boards: stm32: use std openocd config

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview
Use the standard openocd config for STM boards. I've run into issues with openocd and the stm32f429zi board. ~~It's possible this is an improvement.~~ It's probably better to just use the built in boards in openocd so we have fewer files to maintain.

I didn't include the changes to the board that I do have in this PR.

### Testing Strategy

Nothing, unfortunately. I don't have these boards. But I'm worried about bitrot with these boards.

### TODO or Help Wanted

Thoughts?

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by alevy — APPROVED** [review]
I strongly support this. It's very likely that openocd didn't have built in support for these boards originally, but does now. If this breaks something, we can always bring this back.

File: boards/stm32f3discovery/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Licensed under the Apache License, Version 2.0 or the MIT License.
+source [find board/stm32f412g-disco.cfg]; # Licensed under the Apache License, Version 2.0 or the MIT License.
 # SPDX-License-Identifier: Apache-2.0 OR MIT
 # Copyright Tock Contributors 2022.
```
Comment (alevy) [inline boards/stm32f3discovery/Makefile:3]:
```suggestion
# Licensed under the Apache License, Version 2.0 or the MIT License.
# SPDX-License-Identifier: Apache-2.0 OR MIT
# Copyright Tock Contributors 2022.
```

File: boards/stm32f412gdiscovery/Makefile [hunk 2]
```diff
@@ -10,19 +10,18 @@ PLATFORM=stm32f412gdiscovery
 include ../Makefile.common
```
Comment (alevy) [inline boards/stm32f412gdiscovery/Makefile:10]:
```suggestion
# Licensed under the Apache License, Version 2.0 or the MIT License.
# SPDX-License-Identifier: Apache-2.0 OR MIT
# Copyright Tock Contributors 2024.

include ../Makefile.common
```

File: boards/nucleo_f446re/Makefile [hunk 3]
```diff
@@ -5,7 +5,6 @@
 # Makefile for building the tock kernel for the NUCLEO-446RE platform
```
Comment (alevy) [inline boards/nucleo_f446re/Makefile:5]:
```suggestion
# Licensed under the Apache License, Version 2.0 or the MIT License.
# SPDX-License-Identifier: Apache-2.0 OR MIT
# Copyright Tock Contributors 2024.

# Makefile for building the tock kernel for the NUCLEO-446RE platform
```

File: boards/nucleo_f429zi/Makefile [hunk 4]
```diff
@@ -10,19 +10,18 @@ PLATFORM=nucleo_f429zi
 include ../Makefile.common
```
Comment (alevy) [inline boards/nucleo_f429zi/Makefile:10]:
```suggestion
# Licensed under the Apache License, Version 2.0 or the MIT License.
# SPDX-License-Identifier: Apache-2.0 OR MIT
# Copyright Tock Contributors 2024.

include ../Makefile.common
```

**Review by alevy — CHANGES_REQUESTED** [review]
License headers

**Comment by bradjc** [comment]
I don't understand what's happening with the license headers.

**Comment by alevy** [comment]
> I don't understand what's happening with the license headers.

There were some strange diffs going on, and ultimately I don't think the license headers were the problem (but rather an incorrectly removed Makefile variable). But the license headers don't hurt...

**Comment by alevy** [comment]
Oh, I see... I was wrong. Will fix

**Review by alevy — APPROVED** [review]

**Pull request merged**